azext_iot/iothub/params.py

```diff
@@ -246,3 +246,88 @@ def load_iothub_arguments(self, _):
             options_list=["--content-type", "--ct"],
             help="MIME Type of file.",
         )
+
+    with self.argument_context("iot edge devices") as context:
+        context.argument(
+            "devices",
+            options_list=["--device", "-d"],
+            nargs="+",
+            action="append",
+            help="Space-separated key=value pairs corresponding to properties of the edge device to create. "
+            "The following key values are supported: `id`, `deployment` (inline json or path to file), `hostname`, "
+            "`parent`, `edge_agent`, and `container_auth` (inline json or path to file). "
+            "--device can be used 1 or more times. Review help examples for parameter usage."
+        )
+        context.argument(
+            "clean",
+            options_list=["--clean", "-c"],
+            arg_type=get_three_state_flag(),
+            help="Deletes all devices in target hub before creating new devices.",
```

Comment:
this sounds like a command that needs a prompt... This goes back to why can't we just have a delete command

Reply:
Agree, we have talked offline about adding a confirmation to this command if `--clean` is provided.